### PR TITLE
refactor(tests): clean up ActivityHistoryTests and InstructionsEndpoi…

### PR DESCRIPTION
**Pull request by hasith**
…ntTests

- Removed commented-out test execution commands and outdated test methods from ActivityHistoryTests and InstructionsEndpointTests.
- Streamlined the test classes for better readability and maintainability.

These changes enhance the clarity and focus of the integration tests, ensuring they are up-to-date and easier to navigate.

### CODE REVIEW

**Review by hasith — COMMENTED** [review]
## ✅ Automated PR Review — APPROVE

**PR #436 · refactor(tests): clean up ActivityHistoryTests and InstructionsEndpointTests**
**Author:** Hasith Yaggahavita
**Head:** `bc68d414baf24573cdc356badf31ce0a97b5a171`
**Base:** `main` (`9c3c8e670494e31b8f62d13fabb2217f0ae4f66a`)

---

### Summary

This is a pure housekeeping refactor of two integration test files. The changes are entirely cosmetic:

| File | Change |
|------|--------|
| `ActivityHistoryTests.cs` | Removed blank lines, 2× block-comments containing `dotnet test --filter ...` CLI commands, and a trailing comment noting previously removed tests. Fixed trailing whitespace on closing brace. |
| `InstructionsEndpointTests.cs` | Removed blank lines and 3× block-comments containing `dotnet test --filter ...` CLI commands. |

**No logic, test coverage, or behaviour was changed.** All existing `[Fact]` test methods remain intact. The removed comments served only as developer convenience shortcuts for manually running individual tests from the terminal — they are not needed in a properly configured CI environment.

---

### Findings

| Severity | Count |
|----------|-------|
| 🔴 Critical | 0 |
| 🟡 Warning | 0 |
| 🔵 Suggestion | 0 |

**No issues found.** Both correctness/regression and security/edge-case scans returned clean.

---

### Verdict: ✅ APPROVE

This refactor improves readability by reducing visual noise. Safe to merge.

**Review by hasith — COMMENTED** [review]
## PR Review — `refactor(tests): clean up ActivityHistoryTests and InstructionsEndpointTests`

### Re-review delta
Reviewed 1 new commit(s) since the last review (`bc68d414`..`ce2dbfe8`).
- ✅ Fixed: 0 previously-flagged issue(s) resolved
- ⏳ Still open: 0 carried-over issue(s)
- 🆕 New: 2 issue(s) introduced since the last review

---

### Verdict: ✅ APPROVE

This is a well-scoped cleanup PR — 32 test files touched, all changes are whitespace normalisation and removal of stale commented-out code. No production logic is modified. The two findings below are about test coverage gaps that **pre-date this PR** (the missing assertions were already commented-out; the missing test methods were already removed). The cleanup makes these gaps more visible, which is a net positive.

---

### 🟡 Warnings (2)

**1. `GetLatestByName_WithValidNameAndAgent_ReturnsKnowledge` test only verifies the response name — not content or recency**
`XiansAi.Server.Tests/IntegrationTests/WebApi/KnowledgeEndpointsTests.cs:129`

The test creates two knowledge records (`"Old content"` and `"Latest content"`) and then only asserts `result.Name == "test-knowledge"`. The two assertions confirming `result.Content == "Latest content"` and `result.Id == knowledge2.Id` were already commented out before this PR and are now permanently deleted. The test no longer verifies that the `/api/client/knowledge/latest` endpoint actually returns the **most recent** record.

```csharp
// Current state — only name is verified:
Assert.Equal("test-knowledge", result.Name);

// These should be restored:
Assert.Equal("Latest content", result.Content);
Assert.Equal(knowledge2.Id, result.Id);
```

**2. `ActivityHistoryTests` is missing input-validation and persistence tests**
`XiansAi.Server.Tests/IntegrationTests/AgentApi/ActivityHistoryTests.cs:56`

The `// Removed: CreateActivityHistory_WithInvalidData_ReturnsBadRequest` and `// Removed: CreateActivityHistory_VerifyDataInsertedIntoMongoDB` comment markers confirm these two integration tests were previously deleted. The cleanup removes those markers, leaving just one happy-path test. There is no integration coverage for:
- Malformed/invalid request payloads being rejected with `400`
- Submitted activity history actually being persisted to MongoDB

---

### 💡 Suggestions (1)

**Consolidation of `CreateTestMessageAsync` in `MessagingEndpointsTests`** is a good improvement — verify all existing callers pass a `scope` argument explicitly where one was previously assumed.

---

### ✅ What's done well
- Consistent removal of 100+ `dotnet test --filter` comment blocks across 32 files — this was cluttering every test class
- Variable renaming (`scope` → `serviceScope`) to avoid shadowing a parameter — correct and consistent
- `CreateTestMessageAsync` / `CreateTestMessageWithScopeAsync` consolidation reduces duplication
- Proper newline at end-of-file added to `ActivityHistoryTests.cs` and `TestAuthHandler.cs`

File: XiansAi.Server.Tests/IntegrationTests/WebApi/KnowledgeEndpointsTests.cs [hunk 1]
```diff
@@ -113,29 +109,24 @@ public async Task Create_WithValidRequest_CreatesKnowledge()
...
         // Act
         var response = await GetAsync($"/api/client/knowledge/latest?name=test-knowledge&agent={agentName}");
 
         // Assert
         Assert.Equal(HttpStatusCode.OK, response.StatusCode);
-        
+
         var result = await ReadAsJsonAsync<Knowledge>(response);
         Assert.NotNull(result);
         Assert.Equal("test-knowledge", result.Name);
```
Comment (hasith) [inline XiansAi.Server.Tests/IntegrationTests/WebApi/KnowledgeEndpointsTests.cs:129]: ⚠️ **WARNING — Test only verifies name, not content or recency**

This test creates two knowledge records with different content (`"Old content"` vs `"Latest content"`) but only asserts that the response has the correct `Name`. The assertions that would confirm the endpoint returns the **most recent** record were previously commented-out and are now permanently deleted by this PR.

```csharp
// Only this assertion remains — it passes regardless of which version is returned:
Assert.Equal("test-knowledge", result.Name);

// These were removed — they should be restored to make the test meaningful:
// Re-capture the return value:
// var knowledge2 = await CreateTestKnowledgeAsync("test-knowledge", agentName, "Latest content");
// Then assert:
Assert.Equal("Latest content", result.Content);
Assert.Equal(knowledge2.Id, result.Id);
```

Without these assertions, the `GetLatestByName` endpoint could return the **old** knowledge record and this test would still pass.

File: XiansAi.Server.Tests/IntegrationTests/AgentApi/ActivityHistoryTests.cs [hunk 2]
```diff
@@ -53,15 +45,12 @@ public async Task CreateActivityHistory_WithValidData_ReturnsOk()
...
+
         // Wait for background tasks to complete with a reasonable timeout
         await backgroundTaskService.WaitForCompletionAsync(TimeSpan.FromSeconds(5));
     }
-    
-    // Removed: CreateActivityHistory_WithInvalidData_ReturnsBadRequest
-    // Removed: CreateActivityHistory_VerifyDataInsertedIntoMongoDB
-} 
\ No newline at end of file
+}
```
Comment (hasith) [inline XiansAi.Server.Tests/IntegrationTests/AgentApi/ActivityHistoryTests.cs:56]: ⚠️ **WARNING — Activity history tests missing for invalid input and data persistence**

This class now has only one test (`CreateActivityHistory_WithValidData_ReturnsOk`). The deleted `// Removed:` comment markers confirm two previously-existing tests are gone, leaving the activity history endpoint with no integration-level coverage for:

| Missing test | What it verified |
|---|---|
| `CreateActivityHistory_WithInvalidData_ReturnsBadRequest` | Malformed/invalid payloads are rejected with `400 Bad Request` |
| `CreateActivityHistory_VerifyDataInsertedIntoMongoDB` | Submitted activity history records are actually persisted to MongoDB |

Without the persistence test in particular, a regression where the background task silently drops records would not be caught by this suite.

**Recommendation:** Re-add both tests, or add a follow-up ticket to track their restoration.
